server/sonar-web/src/main/js/apps/issues2/components/App.js

```diff
@@ -118,6 +121,95 @@ export default class App extends React.PureComponent {
     this.mounted = false;
   }
 
+  attachShortcuts() {
+    window.addEventListener('keydown', this.handleKeyDown, false);
+  }
+
+  detachShortcuts() {
+    window.removeEventListener('keydown', this.handleKeyDown);
+  }
+
+  handleKeyDown = (event: KeyboardEvent) => {
+    switch (event.keyCode) {
+      case 38:
+        // up
+        event.preventDefault();
+        this.selectPreviousIssue();
+        break;
+      case 40:
+        // down
+        event.preventDefault();
+        this.selectNextIssue();
+        break;
+      case 39:
+        // right
+        event.preventDefault();
+        this.openSelectedIssue();
+        break;
+      case 37:
+        // left
+        event.preventDefault();
+        this.closeIssue();
+        break;
+    }
+  };
+
+  getSelectedIndex(): ?number {
+    const { issues, selected } = this.state;
+    return issues != null && selected != null && issues.includes(selected)
+      ? issues.indexOf(selected)
+      : null;
+  }
+
+  selectNextIssue = () => {
+    const { issues } = this.state;
+    const selectedIndex = this.getSelectedIndex();
+    if (issues != null && selectedIndex != null && selectedIndex < issues.length - 1) {
+      if (getOpen(this.props.location.query)) {
+        this.openIssue(issues[selectedIndex + 1]);
+      } else {
+        this.setState({ selected: issues[selectedIndex + 1] });
```

Comment:
You set the state based on `issues` that is part of the state. I think in this situation it's better to use `this.setState(prevState =>...)` ?

But I'm not sure you can use it easily in this case, since you don't always set the state.